### PR TITLE
Finish Booster

### PR DESCRIPTION
**Pull request by JaceCear**

### CODE REVIEW

File: src/interactable_booster.c [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+        gPlayer.unk14 = BOOSTER_SPEED;
+
+void initSprite_Interactable_Booster(Interactable *ia, u16 spriteRegionX,
+                                     u16 spriteRegionY, u8 spriteY)
+{
+    struct Task *t = TaskCreate(Task_Interactable_Booster, sizeof(Sprite_Booster),
+                                0x2000, 0, TaskDestructor_80095E8);
+    Sprite_Booster *booster = TaskGetStructPtr(t);
+    struct UNK_0808B3FC_UNK240 *main = &booster->main;
+    int value;
```
Comment (freshollie) [inline src/interactable_booster.c:50]:
```suggestion
    u32 value;
```

I wouldn't use generic ints, even if the compiler doesn't really care about the size in this example

File: src/interactable_booster.c [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    if (gPlayer.unk14 < BOOSTER_SPEED)                                                  \
+        gPlayer.unk14 = BOOSTER_SPEED;
+
+void initSprite_Interactable_Booster(Interactable *ia, u16 spriteRegionX,
+                                     u16 spriteRegionY, u8 spriteY)
+{
+    struct Task *t = TaskCreate(Task_Interactable_Booster, sizeof(Sprite_Booster),
+                                0x2000, 0, TaskDestructor_80095E8);
+    Sprite_Booster *booster = TaskGetStructPtr(t);
+    struct UNK_0808B3FC_UNK240 *main = &booster->main;
```
Comment (freshollie) [inline src/interactable_booster.c:49]: Btw, still not sure about the `main` convention, this is the Sprite which will be drawn to screen, `screenSprite`?

File: src/interactable_booster.c [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+extern void TaskDestructor_80095E8(struct Task *);
+
+extern bool32 sub_800C204(struct UNK_0808B3FC_UNK240 *, s16, s16, s16,
+                          struct SomeStruct_59E0 *, u32);
+extern void sub_80218E4(struct SomeStruct_59E0 *);
+extern void sub_8023B5C(struct SomeStruct_59E0 *, s8);
+
+extern const struct UNK_080E0D64 gUnknown_080D942C[2][6];
+
+#define BOOSTER_SPEED 3072
```
Comment (freshollie) [inline src/interactable_booster.c:29]: Cool we can now set this 😎

Comment (JaceCear) [reply src/interactable_booster.c:29]: Not completely freely while the whole game doesn't match, but yeah! :D

**Review by freshollie — APPROVED** [review]

**Pull request merged**